iot_config/src/route_service.rs

```diff
@@ -390,15 +383,13 @@ impl iot_config::Route for RouteService {
 
         tracing::info!("client subscribed to route stream");
         let pool = self.pool.clone();
-        let shutdown_listener = self.shutdown.clone();
         let (tx, rx) = tokio::sync::mpsc::channel(20);
         let signing_key = self.signing_key.clone();
 
         let mut route_updates = self.subscribe_to_routes();
 
         tokio::spawn(async move {
             tokio::select! {
-                _ = shutdown_listener.clone() => return,
```

Comment:
this select statement can just go away if we're not selecting between two futures and the stream of combinators can just be awaited with a map_err at the end. that's what this used to do before i tried selecting on the shutdown to attempt to exit more gracefully

Reply:
addressed here: ec80d34b0873a8a45fb277efa7f5b64b8cb28bf8